_posts/en/newsletters/2020-03-04-newsletter.md

```diff
@@ -0,0 +1,226 @@
+layout: newsletter
+lang: en
+---
+This week's newsletter describes a proposed update to BIP340 schnorr
+keys and signatures, seeks feedback on a proposal to improve startup
+feature negotiation between full nodes, examines a suggestion for a
+standardized way to prevent hardware wallets from using corrupt nonces
+to leak private keys, links to an analysis of the properties necessary
+in a hash function for taproot to be secure, and summarizes recent
+topics from the Bitcoin Core PR Review Club.  Also included are our
+regular sections for release announcements and notable changes to
+popular Bitcoin infrastructure projects.
+
+## Action items
+
+*None this week.*
+
+## News
+
+- **Updates to BIP340 schnorr keys and signatures:** as previously
+  mentioned in two separate items in Newsletter #84 ([1][news83 safety], [2][news83 tiebreaker]),
+  several updates to [BIP340][] schnorr signatures have been proposed
+  (which also affect [BIP341][] taproot).  Pieter Wuille [suggests][wuille
+  update] changing which variant of a public key should be used with
+  [BIP340][], with the new selection being based on the eveness of the
+  key.  Also, the recommended procedure for generating a nonce has been
+  changed to include the public key in nonce generation, to include
+  extra randomness when available, and to include a step in the nonce
+  generation algorithm that obfuscates the private key by the randomness
+  in order to protect the key against [differential power analysis][].
+
+    Because of the significance of some of these changes, the tag for
+    the tagged hashes in BIP340 is changed, ensuring that any code
+    written for the earlier drafts will generate signatures that fail
+    validation under the proposed revisions.  Wuille requests community
+    feedback on the changes.
+
+- **Improving feature negotiation between full nodes at startup:** Suhas
+  Daftuar [requested feedback][daftuar wtxid] on a proposal to insert a
+  message into the sequence used by nodes to open a connection with a
+  new peer.  The new message would make it easier for a node to
+  negotiate what features it wants to receive from its peer.  A
+  challenge here is that previous versions of Bitcoin Core would
+  terminate a new connection if certain messages didn't appear in a
+  particular order---and it's into this strict sequence that Daftuar
+  wants to insert a new message.  The proposal does increment the P2P
+  protocol version to provide backwards compatibility, but Daftuar is
+  seeking feedback from maintainers of full nodes about whether the
+  insertion of negotiation messages would cause any problems.  If you're
+  aware of any problems, please reply to the thread.
+
+- **Proposal to standardize an anti covert nonce protocol:** Stepan
+  Snigirev [started][snigirev nonce] a discussion on the Bitcoin-Dev
+  mailing list about standardizing a protocol that prevents hardware
+  wallets from generating a signature that uses a biased nonce to leak
```

Comment:
> I think you'd need a series of biased nonces to leak information

Definitely not my expertise, but I assume one biased nonce would be enough to recover the individual private key if you knew enough about the algorithm used to create the bias (e.g. what if the bias was always `nonce = 0`).  I think leaking the wallet seed, master private key, or the chain code corresponding to a private key would probably take at least two signatures/nonces (and more than that if you wanted to prevent third parties from detecting the exfiltration or using it to their own advantage before you).

Reply:
I'm also not an expect. I think the new wording ("using biased nonces") is safer though!